### ISSUE TITLE
It Does Not Work

### ISSUE THREAD

**Issue by StevenLeonCooper** [dotnet/AspNetCore.Docs#30010]
I followed all the steps but when I got to "Test the web API" I just got nothing, a blank array for /Books and "not found" when I tried a specific book ID. 

The DB connection info appears correct and I can see my Bookstore database and Books table in Compass and the sample data is exactly as suggested. 

So how do you troubleshoot when there's no real error message that would help me figure out why the API returns a blank array instead of a list of my books? 

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 5240866d-d0a1-f0dd-7b2c-203e1d69bc4b
* Version Independent ID: 83a5cb3b-23bc-4f0c-cd1c-1d1ffeb1131a
* Content: [Create a web API with ASP.NET Core and MongoDB](https://learn.microsoft.com/en-us/aspnet/core/tutorials/first-mongo-app?view=aspnetcore-7.0&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/first-mongo-app.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/first-mongo-app.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @wadepickett
* Microsoft Alias: **wpickett**

**Comment by Rick-Anderson** [comment]
Hello! Thanks for contacting us. This sounds like a general question about using ASP.NET Core. While we try to look at and respond to all issues, for questions like this we recommend posting to a community support group like [Stack Overflow](https://stackoverflow.com) with the `asp.net-core` tag.

* [Stack Overflow](https://stackoverflow.com/questions)
* [ASP.NET Core Slack](http://tattoocoder.com/aspnet-slack-sign-up/)
* [ASP.NET Gitter](https://gitter.im/aspnet/Home)

Potential bugs in the ASP.NET Core framework and product feedback can be reported at [dotnet/aspnetcore issues](https://github.com/dotnet/aspnetcore/issues) (include full repro steps and cross-link to a repro project in GH when appropriate). For Visual Studio, use the [Report a Problem](https://docs.microsoft.com/visualstudio/ide/how-to-report-a-problem-with-visual-studio) gesture or [Suggest a Feature](https://docs.microsoft.com/visualstudio/ide/suggest-a-feature?view=vs-2019) gesture within VS, which open internal issues for the VS team. For Visual Studio Code, you can open an issue at [microsoft/vscode issues](https://github.com/Microsoft/vscode/issues).

**Comment by wowTerrific** [comment]
> I followed all the steps but when I got to "Test the web API" I just got nothing, a blank array for /Books and "not found" when I tried a specific book ID.
> 
> The DB connection info appears correct and I can see my Bookstore database and Books table in Compass and the sample data is exactly as suggested.
> 
> So how do you troubleshoot when there's no real error message that would help me figure out why the API returns a blank array instead of a list of my books?
> 
> #### Document Details
> ⚠ _Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking._
> 
> * ID: 5240866d-d0a1-f0dd-7b2c-203e1d69bc4b
> * Version Independent ID: 83a5cb3b-23bc-4f0c-cd1c-1d1ffeb1131a
> * Content: [Create a web API with ASP.NET Core and MongoDB](https://learn.microsoft.com/en-us/aspnet/core/tutorials/first-mongo-app?view=aspnetcore-7.0&tabs=visual-studio)
> * Content Source: [aspnetcore/tutorials/first-mongo-app.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/first-mongo-app.md)
> * Product: **aspnet-core**
> * Technology: **aspnetcore-tutorials**
> * GitHub Login: @wadepickett
> * Microsoft Alias: **wpickett**

I know this is old, but I had the same issue. The tutorial seems to be missing a pretty crucial step between creating the controller and testing the app. I had to add the following lines to `Program.cs` in order for it to work:
`builder.Services.AddControllers();`
`app.MapControllers();`

**Comment by StevenLeonCooper** [comment]
> > I followed all the steps but when I got to "Test the web API" I just got nothing, a blank array for /Books and "not found" when I tried a specific book ID.
> > The DB connection info appears correct and I can see my Bookstore database and Books table in Compass and the sample data is exactly as suggested.
> > So how do you troubleshoot when there's no real error message that would help me figure out why the API returns a blank array instead of a list of my books?
> > #### Document Details
> > ⚠ _Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking._
> > 
> > * ID: 5240866d-d0a1-f0dd-7b2c-203e1d69bc4b
> > * Version Independent ID: 83a5cb3b-23bc-4f0c-cd1c-1d1ffeb1131a
> > * Content: [Create a web API with ASP.NET Core and MongoDB](https://learn.microsoft.com/en-us/aspnet/core/tutorials/first-mongo-app?view=aspnetcore-7.0&tabs=visual-studio)
> > * Content Source: [aspnetcore/tutorials/first-mongo-app.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/first-mongo-app.md?rgh-link-date=2024-01-31T15%3A17%3A18Z)
> > * Product: **aspnet-core**
> > * Technology: **aspnetcore-tutorials**
> > * GitHub Login: @wadepickett
> > * Microsoft Alias: **wpickett**
> 
> I know this is old, but I had the same issue. The tutorial seems to be missing a pretty crucial step between creating the controller and testing the app. I had to add the following lines to `Program.cs` in order for it to work: `builder.Services.AddControllers();` `app.MapControllers();`

Yes I noticed a few missing steps. If you think that's bad, try following the instructions for Mac OS. They straight-up forgot they were writing instructions for a different operating system in the middle and left in a bunch of instructions on modifying the PATH environment variable to run MongoDB commands. MacOS doesn't even have environment variables.

**Comment by wadepickett** [comment]
Thanks for letting us know this topic needs help and your experiences with it.  We will get it fixed!  Looks like a section must be accidentally outside the Visual Studio Windows tab area for starters and a missing instruction.  Re-opening this issue.
